### PR TITLE
Added outline documentation, and details of data licenses.

### PR DESCRIPTION
**Pull request by andreweland**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+At a high level, the coursework flow is:
+* Train a model to detect acute kidney injury from a dataset of synthetic blood test results, supplied as a CSV file ([example spec](https://docs.google.com/document/d/13zLCj8fCinawuR5kUD5NsHrPobnQvhfCMnMCYAdhv4I/edit#heading=h.nhjmk7sgdyby)).
+* Use the trained model to build a system that predicts acute kidney injury in real time, using blood test results supplied as [HL7 messages](https://en.wikipedia.org/wiki/Health_Level_7#HL7_Version_2) over an [MLLP connection](https://hl7.skyware-group.com/lib/exe/fetch.php?media=wiki:mllp.pdf).
+* Add the functionality necessary to ensure their system is robust in the presence of environmental failures.
+* Add the monitoring necessary to ensure groups can detect and respond to system failures.
+* Run their system in a simulated environment over a period of two weeks, in the presence of deliberate environmental failures (eg machine restarts, corrupt data).
```
Comment (xprogramari) [inline README.md]: nit / run `the` system

File: generator/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Software Engineering for Machine Learning Systems
+
+## Generator
+
+This directory contains code to generate synthetic creatinine blood test results, acute kidney injury events, and patient admit and discharge within a hospital. It makes no attempt to be clinically accurate, but beleivable enough to reason about when building a system.
```
Comment (xprogramari) [inline generator/README.md]: typo / believable

File: generator/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+A random set of patients is selected to suffer acute kidney injury, and the blood test results for these patients are constructed such that the [NHS reference AKI algorithm](https://www.england.nhs.uk/akiprogramme/aki-algorithm/) performs well. Bias is deliberately added to enable the design of simple machine learning models with better performance.
+
+
+Patient ages and sexes are reasonably distributed, as are their blood result values. Event times and frequencies bear some resemblance to what you'd see in a reasonbly sized hospital.
```
Comment (xprogramari) [inline generator/README.md]: typo / reasonably (sized hospital)

File: generator/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+The generator can be run with:
+
+```
+pip3 install -r ../requirements.txt
+./generator.py --data=../data --history=90 --days=30 --output=.
+```
+* `--data` specifies the location of the census and name datasets used to generate the synthetic population.
+* `--history` specifies the number of days of patient history that should be generated, written as a CSV dataset of historical blood test results.
+* `--days` specific the number of hospital events that should be generated, written as an MLLP encoding set of HL7 messages.
```
Comment (xprogramari) [inline generator/README.md]: last two (days/ output) specifies / typos 
but also u can just omit all specifies imo

File: generator/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+* `--output` specified the directory in which output files should be written.
+
+The generator produces these outputs:
+* `messages.mllp` an MLLP encoded sequence of events at the hospital over the given number of days. It includes:
+    * `ADT^A01` events when a patient is admitted. These messages include patient demographic data.
```
Comment (xprogramari) [inline generator/README.md:25]: i m being really nitty here but events when feels weird
maybe events for admitting a patient / events for recording blood test results etc

File: generator/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* `history.csv` a CSV file with the historical blood test results for patients who may appear in `messages.mllp`. Systems receiving the generated HL7 messages will need to take the blood test results from this file into account to generate accurate predictions.
+* `training.csv` a CSV file with a dataset that can be used to train a model to detect acute kidney injury. For each patient, the file includes demographics, blood test result values, and whether they were diagonsed with an acute kidney injury. If the `aki` column is `y` for a patient, then their most recent blood test result represents the point at which that diagonsis was made. No further blood test results are included post diagonsis.
+
+  Note that the same event stream is used to create both this output file, and the others, so if `training.csv` is used, the other files shouldn't be; fresh data should be generate instead.
```
Comment (xprogramari) [inline generator/README.md]: typo / generated

File: simulator/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+groups:
+- name: mimico
+  mllp: 9440
+  pager: 10440
+- name: volantin
+  mllp: 9441
+  pager: 10441
+
+```
+In this case, the simulator is serving two different groups, named `mimico` and `volantin`, using ports 9440 and 9441 to listen for MLLP connections from the respective groups, and 10440 and 10441 for pager requests. Using group names inspired by the [bridges of architect Santiago Calatrava](https://calatrava.com/projects.html?project_type=bridges) is encouraged, but not required.
```
Comment (xprogramari) [inline simulator/README.md:44]: cute bridges

**Comment by xprogramari** [comment]
lgtm / surprisingly few typos

File: README.md (hunk 1 above)
Comment (andreweland) [reply README.md]: Done.

File: generator/README.md (hunk 2 above)
Comment (andreweland) [reply generator/README.md]: Done.

File: generator/README.md (hunk 3 above)
Comment (andreweland) [reply generator/README.md]: Done.

File: generator/README.md (hunk 4 above)
Comment (andreweland) [reply generator/README.md]: I removed all the specifies.

File: generator/README.md (hunk 5 above)
Comment (andreweland) [reply generator/README.md:25]: I switched it to _an MLLP encoded sequence of HL7 message representing events_

File: generator/README.md (hunk 6 above)
Comment (andreweland) [reply generator/README.md]: Done.

File: simulator/README.md (hunk 7 above)
Comment (andreweland) [reply simulator/README.md:44]: Aren't they?? My favourite is [Serreria](https://calatrava.com/projects/serreria-bridge-valencia.html).

**Comment by andreweland** [comment]
Thanks for reading!

**Pull request merged**